htdocs/mrp/class/mo.class.php

```diff
@@ -612,11 +612,6 @@ public function update(User $user, $notrigger = false)
 			$error++;
 		}
 
-		$result = $this->updateProduction($user, $notrigger);
```

Comment:
I think usage of this method here is done on purpose.
However, inside this method it seems to miss code under the comment 

// TODO Check that production has not started. If yes, we stop here.

There is nothing in this TODO. We call the method when we are in draft mode, but we should do nothing if production has started.

Reply:
I admit that I don't understand the need to delete the MO lines to add them later (without the changes made before the MO update).
Indeed, I understand the use of an update function for MO lines: It is possible to modify the quantity of an OF in "Draft" status, its lines must therefore be modified accordingly. However, this does not require deleting the rows and then adding them again.

Moreover, as I said above, when I create an MO without an associated BOM, add lines to it and then modify the MO (the label for example), all the lines previously entered are deleted. And it has nothing to do with whether production has started or not. This is due to the fact that we delete all the lines of the MO to add that of the associated BOM, but what happens to MOs that do not have associated BOMs?

I think the solution is to use a real MO rows update function (like it's done for all other objects in Dolibarr)

I'm developing a fix to correct that